bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java

```diff
@@ -336,6 +328,44 @@ public Collection<Rule> getByTags(String... tags) {
         return result;
     }
 
+    @Override
+    public void regenerateFromTemplate(String ruleUID) {
+        Rule rule = get(ruleUID);
+        if (rule == null) {
+            throw new IllegalArgumentException(
+                    "Can't regenerate rule from template because no rule with UID \"" + ruleUID + "\" exists");
+        }
+        if (rule.getTemplateUID() == null || rule.getTemplateState() == TemplateState.NO_TEMPLATE) {
+            throw new IllegalArgumentException(
+                    "Can't regenerate rule from template because the rule isn't linked to a template");
+        }
+        try {
+            Rule resolvedRule = resolveRuleByTemplate(
+                    RuleBuilder.create(rule).withActions((List<Action>) null).withConditions((List<Condition>) null)
+                            .withTriggers((List<Trigger>) null).withTemplateState(TemplateState.PENDING).build());
+            Provider<Rule> provider = getProvider(rule.getUID());
+            if (provider == null) {
+                logger.error("Regenerating rule '{}' from template failed because the provider is unknown",
+                        rule.getUID());
+                return;
+            }
+            if (provider instanceof ManagedRuleProvider) {
+                update(resolvedRule);
+            } else {
+                updated(provider, rule, resolvedRule);
+            }
+            if (resolvedRule.getTemplateState() == TemplateState.TEMPLATE_MISSING) {
+                logger.warn("Failed to regenerate rule '{}' from template since the template is missing",
+                        rule.getUID());
+            } else {
+                logger.info("Rule '{}' was regenerated from template '{}'", rule.getUID(), rule.getTemplateUID());
```

Comment:
This should better only be debug.

Reply:
This is the result of a user action, which is why I thought it would be nice to see a confirmation. But, I can change it if you think that's the best way.

Reply:
Ok, as long as there is no automatic or bulk regeneration possible, then it should be ok.